### ISSUE TITLE
%load doesn't work in notebook or console

### ISSUE THREAD

**Issue by fperez** [jupyter/jupyterlab#581]
Typing `%load <file-or-url>` in the notebook or console should replace the current input with the contents and leave the `load` line commented out, e.g. this:

![image](https://cloud.githubusercontent.com/assets/57394/17259539/dcaa0e0c-5580-11e6-8e24-b987cdfdf4ff.png)

becomes:

![image](https://cloud.githubusercontent.com/assets/57394/17259549/e6118d8a-5580-11e6-9bf0-0a972e7364eb.png)

That uses a special api call to reset the current input, it probably isn't implemented yet.

**Comment by blink1073** [comment]
Dev note: we need to respect the `set_next_input` pager data in execution results: 
http://jupyter-client.readthedocs.io/en/latest/messaging.html#payloads-deprecated

**Comment by blink1073** [comment]
This should apply to the console as well.

**Comment by fperez** [comment]
On Fri, Jul 29, 2016 at 12:00 PM, Steven Silvester <notifications@github.com
> wrote:

> This should apply to the console as well.
>
correct, it currently does work in the qtconsole, so the feature parity is
missing for both the notebook and the console.


-- 
Fernando Perez (@fperez_org; http://fperez.org)
fperez.net-at-gmail: mailing lists only (I ignore this when swamped!)
fernando.perez-at-berkeley: contact me here for any direct mail

**Comment by fperez** [comment]
Works like a charm in the console, thx!

In the nb it's slightly sub-ideal, but actually on-par with the current notebook: the focus after `%load` is on the *next* cell, not the current one where the data gets loaded.

Do you think it's worth tracking that as an enhancement, or do you think that special-casing the focus management for this particular use case would be too much of a hassle?

Basically, UX-wise it's quite natural to `%load foo` and then to be positioned right where `foo` arrived (i.e. on the same cell) to either edit or execute its contents.  That's what happens now in the console.

But it would require doing something funny when the current cell is overwritten, since the natural focus flow is to go to the next cell after execution.

In summary: I'm OK with not making this small improvement if it's more hassle than it's worth.  But if you think it's easy/worthwhile, then go for it (and we can open a new issue for it that would be purely enhancement/long term, 1.0 or later).

**Comment by jasongrout** [comment]
We already have this special case when there is an `input()`. Makes sense to have it here too.

**Comment by fperez** [comment]
Excellent, thx! Will continue on #678 as needed.
